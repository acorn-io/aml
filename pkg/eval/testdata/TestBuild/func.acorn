x: x(a,b)
y: x(a, b, c: 3, d: 4)