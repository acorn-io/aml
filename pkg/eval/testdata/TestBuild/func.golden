&eval.File{Body: &eval.Struct{
	Position: eval.Position{
		Filename: "func.acorn",
		Line:     1,
		Column:   1,
	},
	Fields: []eval.Field{
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{
					Value: value.String("x"),
				},
				Pos: eval.Position{
					Filename: "func.acorn",
					Line:     1,
					Column:   1,
				},
			},
			Value: &eval.Call{
				Pos: eval.Position{
					Filename: "func.acorn",
					Offset:   4,
					Line:     1,
					Column:   5,
				},
				Func: &eval.Lookup{
					Pos: eval.Position{
						Filename: "func.acorn",
						Offset:   3,
						Line:     1,
						Column:   4,
					},
					Key: "x",
				},
				Args: []eval.Field{
					&eval.Embedded{
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   5,
							Line:     1,
							Column:   6,
						},
						Expression: &eval.Lookup{
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   5,
								Line:     1,
								Column:   6,
							},
							Key: "a",
						},
					},
					&eval.Embedded{
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   7,
							Line:     1,
							Column:   8,
						},
						Expression: &eval.Lookup{
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   7,
								Line:     1,
								Column:   8,
							},
							Key: "b",
						},
					},
				},
			},
			Pos: eval.Position{
				Filename: "func.acorn",
				Line:     1,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("y")},
				Pos: eval.Position{
					Filename: "func.acorn",
					Offset:   10,
					Line:     2,
					Column:   1,
				},
			},
			Value: &eval.Call{
				Pos: eval.Position{
					Filename: "func.acorn",
					Offset:   14,
					Line:     2,
					Column:   5,
				},
				Func: &eval.Lookup{
					Pos: eval.Position{
						Filename: "func.acorn",
						Offset:   13,
						Line:     2,
						Column:   4,
					},
					Key: "x",
				},
				Args: []eval.Field{
					&eval.Embedded{
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   15,
							Line:     2,
							Column:   6,
						},
						Expression: &eval.Lookup{
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   15,
								Line:     2,
								Column:   6,
							},
							Key: "a",
						},
					},
					&eval.Embedded{
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   18,
							Line:     2,
							Column:   9,
						},
						Expression: &eval.Lookup{
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   18,
								Line:     2,
								Column:   9,
							},
							Key: "b",
						},
					},
					&eval.KeyValue{
						Key: eval.FieldKey{
							Key: eval.Value{Value: value.String("c")},
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   21,
								Line:     2,
								Column:   12,
							},
						},
						Value: eval.Value{Value: value.Number("3")},
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   21,
							Line:     2,
							Column:   12,
						},
					},
					&eval.KeyValue{
						Key: eval.FieldKey{
							Key: eval.Value{Value: value.String("d")},
							Pos: eval.Position{
								Filename: "func.acorn",
								Offset:   27,
								Line:     2,
								Column:   18,
							},
						},
						Value: eval.Value{Value: value.Number("4")},
						Pos: eval.Position{
							Filename: "func.acorn",
							Offset:   27,
							Line:     2,
							Column:   18,
						},
					},
				},
			},
			Pos: eval.Position{
				Filename: "func.acorn",
				Offset:   10,
				Line:     2,
				Column:   1,
			},
		},
	},
}}
